examples/text/plot_document_clustering.py

```diff
@@ -3,284 +3,310 @@
 #         Lars Buitinck
+#         Arturo Amor <david-arturo.amor-quiroz@inria.fr>
 # License: BSD 3 clause
 
+# %%
+# Load Data
+# =========
+#
+# We load data from :ref:`20newsgroups_dataset`, which comprises around 18,000
+# newsgroups posts on 20 topics. For illustrative purposes and to reduce the
+# computational cost, we select a subset of 4 topics only accounting for around
+# 3,400 documents. See the example
+# :ref:`sphx_glr_auto_examples_text_plot_document_classification_20newsgroups.py`
+# to gain intuition on the overlap of such topics.
+#
+# Notice that, by default, the text samples contain some message metadata such
+# as `"headers"`, `"footers"` (signatures) and `"quotes"` to other posts. We use
+# the `remove` parameter from :func:`~sklearn.datasets.fetch_20newsgroups` to
+# strip those features and have a more sensible clustering problem.
+
+import numpy as np
 from sklearn.datasets import fetch_20newsgroups
-from sklearn.decomposition import TruncatedSVD
-from sklearn.feature_extraction.text import TfidfVectorizer
-from sklearn.feature_extraction.text import HashingVectorizer
-from sklearn.feature_extraction.text import TfidfTransformer
-from sklearn.pipeline import make_pipeline
-from sklearn.preprocessing import Normalizer
-from sklearn import metrics
 
-from sklearn.cluster import KMeans, MiniBatchKMeans
+categories = [
+    "alt.atheism",
+    "talk.religion.misc",
+    "comp.graphics",
+    "sci.space",
+]
 
-import logging
-from optparse import OptionParser
-import sys
-from time import time
+dataset = fetch_20newsgroups(
+    remove=("headers", "footers", "quotes"),
+    subset="all",
+    categories=categories,
+    shuffle=True,
+    random_state=42,
+)
 
-import numpy as np
+labels = dataset.target
+true_k = np.unique(labels).shape[0]
 
+print(f"{len(dataset.data)} documents - {true_k} categories")
 
-# Display progress logs on stdout
-logging.basicConfig(level=logging.INFO, format="%(asctime)s %(levelname)s %(message)s")
+# %%
+# Comparing Clustering
+# ====================
+#
+# Here we define a function to score different clustering pipelines
+# using several metrics.
+#
+# Given the knowledge of the ground truth class assignments of the samples, it
+# is possible to define metrics to test for different properties of a cluster:
+#
+# - homogeneity, which quantifies cluster containing only members of a single
+#   class;
+#
+# - completeness, which quantifies cluster having members of a given class are
+#   assigned to the same cluster;
+#
+# - V-measure, the harmonic mean of completeness and homogeneity;
+#
+# - Rand-Index, which measures the similarity between pairs of clusters;
+#
+# - Adjusted Rand-Index, a change-adjusted Rand-Index.
+#
+# If the ground truth labels are not known, evaluation can only be performed
+# using the model results itself. In that case, the Silhouette Coefficient
+# comes in handy.
+#
+# For more reference, see :ref:`_clustering_evaluation`.
```

Comment:
On the rendering, the reference to `_clustering_evaluation` is not working properly.